src/registry/lib.rs

```diff
@@ -146,7 +152,7 @@ impl Registry {
         let mut body = ChainedReader::new(vec![Box::new(header) as Box<Reader>,
                                                Box::new(tarball) as Box<Reader>].into_iter());
 
-        let url = format!("{}/api/v1/crates/new", self.host);
+        let url = format!("{}/api/v2/crates/new", self.host);
```

Comment:
I don't think it will be worth it to bump the entire version for this change, it can technically be a largely backwards-compatible extension. I think that we'll want to stick to `v1` for now.